ci-build.sh

```diff
@@ -1,63 +1,73 @@
 #!/usr/bin/env bash
-set -e
 
 FLINK_VERSION=1.8.0
+FLINK_BASE_URL="$(curl -s https://www.apache.org/dyn/closer.cgi\?preferred\=true)flink/flink-${FLINK_VERSION}/"
+FLINK_DIST_FILE_NAME="flink-${FLINK_VERSION}-src.tgz"
+DOWNLOAD_FLINK_URL=${FLINK_BASE_URL}${FLINK_DIST_FILE_NAME}
 
-SCRIPT=$(basename ${BASH_SOURCE[0]})
-ROOT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd -P )"
-
-cd "$ROOT_DIR"
-
-help() {
-  cat <<EOF
-Build Flink Docker images with integrated Prometheus exporter for metrics. Push them to Docker Hub.
-$0 [-h|--help] [-n|--no-push]
-where:
--h | --help     Show this message and exit
--n | --no-push  Don't push the Docker images to Lightbend's Docker Hub account
-EOF
-}
-
-do_push=true
-while [[ $# -gt 0 ]]
-do
-  case $1 in
-    -h|--h*)
-      help
-      exit 0
-      ;;
-    -n|--no*)
-      do_push=false
-      ;;
-    -*)
-      echo "$0: ERROR: Unrecognized option $1"
-      help
-      exit 1
-      ;;
-  esac
-  shift
-done
-
-dockerfiles=(
-  Dockerfile
-  Dockerfile_scala12
-  Dockerfile_alpine
-  Dockerfile_alpine_scala12
-  Dockerfile_ubuntu
-  Dockerfile_ubuntu_scala12)
-
-declare -A suffixes
-
-suffixes[Dockerfile]="scala_2.11"
-suffixes[Dockerfile_scala12]="scala_2.12"
-suffixes[Dockerfile_alpine]="scala_2.11-alpine"
-suffixes[Dockerfile_alpine_scala12]="scala_2.12-alpine"
-suffixes[Dockerfile_ubuntu]="scala_2.11-ubuntu"
-suffixes[Dockerfile_ubuntu_scala12]="scala_2.12-ubuntu"
-
-for f in ${dockerfiles[@]}
-do
-  suffix=${suffixes[$f]}
-  $NOOP docker build -f $f -t lightbend/flink:$FLINK_VERSION-$suffix .
-  $do_push && $NOOP docker push lightbend/flink:$FLINK_VERSION-$suffix
-done
+# Directory for loading src
+TMPDIR="flink_src"
+
+# Directory for build version
+DISTRO="flink"
+
+rm -r ${TMPDIR}
+rm -r ${DISTRO}
+mkdir -p "${TMPDIR}"
+mkdir -p "${DISTRO}"
+chmod 777 "${TMPDIR}"
+chmod 777 "${DISTRO}"
+
+CURL_OUTPUT="${TMPDIR}/${FLINK_DIST_FILE_NAME}"
+
+curl ${DOWNLOAD_FLINK_URL} --output ${CURL_OUTPUT}
+
+# Untar source file and remove it
+
+tar xvzf ${CURL_OUTPUT} -C ${TMPDIR}
+rm ${CURL_OUTPUT}
+
+# Update code
+yes | cp -rf src/main/java/* ${TMPDIR}/flink-${FLINK_VERSION}/flink-runtime/src/main/java
```

Comment:
!! This isn't a viable approach for something people should run in production, as it "hacks" the source code with a patch in an unusual way and then builds.

I think you need to refactor the HA support as follows:
1. Move the Flink code changes to a fork of the Flink project (which should be proposed as a PR to Flink), then build artifacts from that fork.
2. Have this build repo only do the minimum amount of work to build images from that fork with Prometheus and enabling this HA approach.

Reply:
The Flink build needs a custom version string, too, like `1.8.0-lightbend-2.2.0` or something.